spec/integration/working_file_path_spec.rb

```diff
@@ -0,0 +1,298 @@
+require 'rails_helper'
+
+# MasterFile#working_file_path has been a source of problems since it was introduced in 6.4.3
+# This spec file is meant to thoroughly test it in order to find any remaining bugs and to
+# ensure the intended functionality doesn't get broken accidentally in the future.
+#
+# Need to test all of the possiblilites:
+# batch ingest with single file
+# batch ingest with single file skip transcoding
+# batch ingest with pre-transcoded derivatives
+# web upload
+# web upload skip transcoding
+# BE dropbox upload
+# BE dropbox skip transcoding
+# Repeat all of these with and without media path set.
```

Comment:
One thing that I haven't been able to find in the documentation: under what situations will media path be set, and when will it not be set? Might be worth a comment (somewhere, not necessarily here) about why it will or will not be set.

Reply:
:+1: I think we definitely need to add more documentation for guidance around setting/not setting `Settings.matterhorn.media_path`